extensions/grok/deployment/src/main/java/org/apache/camel/quarkus/component/grok/deployment/GrokProcessor.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.grok.deployment;
+
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageResourceBuildItem;
+
+class GrokProcessor {
+
+    private static final String FEATURE = "camel-grok";
+
+    @BuildStep
+    FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    NativeImageResourceBuildItem registerDefaultPatternsNativeResources() {
+        return new NativeImageResourceBuildItem("patterns/patterns");
```

Comment:
I heart of grok for the first time today, but anyway, isn't `patterns/patterns` a typo?

Regardless whether it is correct, wouldn't it be worthwhile to document path?

Shouldn't the itest cover it somehow (I cannot see `patterns/patterns` folder in the itest)?

Reply:
Thanks for raising @ppalaga.

camel-grok comes with what the [camel doc](https://camel.apache.org/components/latest/dataformats/grok-dataformat.html#_predefined_patterns) calls `predefined patterns`, under the hood it's [java-grok default patterns](https://github.com/thekrakken/java-grok/blob/master/src/main/resources/patterns/patterns). Grok default patterns includes things like mac, ip, number and so on.

That said, `patterns/patterns` may be a bit vague but is not under our control. The test coverage must be correct as we are using mac, ip and so on in itests. Concerning the doc, I would let it in camel, maybe just clarify the predefined patterns section.

WDYT ?

Reply:
Oh, I see, so, `patterns/patterns` is not intended to be used by end users, right? All is fine then. Thanks for explaining.

Reply:
Not for end user indeed. Thanks for asking, this exchange will help me to improve the camel doc a bit and may help users in the future.